src/domain/events/edit/EventEdit.tsx

```diff
@@ -25,13 +27,40 @@ const EventEdit = (props: any) => {
   const [selectedLanguage, selectLanguage] = useState(Language.FI);
   const translation = `translations.${selectedLanguage}`;
 
+  // Stolen https://github.com/marmelab/react-admin/issues/2077#issuecomment-516821629
+  // FIXME: Is it really needed?
+  function formatImage(value: any) {
+    if (!value.image || typeof value.image === 'string') {
+      // Value is null or the url string from the backend, wrap it in an object so the form input can handle it
+      return { image: value };
+    } else {
+      // Else a new image is selected which results in a value object already having a preview link under the url key
+      return value.image;
+    }
+  }
+
   return (
     <Edit title={translate('events.edit.title')} {...props}>
       <SimpleForm redirect="show" validate={validateVenue}>
         <LanguageTabs
           selectedLanguage={selectedLanguage}
           onSelect={selectLanguage}
         />
+        <ImageInput
+          format={formatImage}
+          source="image"
+          label="Event image"
+          accept="image/*"
+          placeholder={<p>Drop your file here</p>}
+        >
+          <ImageField source="image" />
+        </ImageInput>
+        <TextInput
+          source={`${translation}.imageAltText`}
+          label={translate('events.fields.imageAltText.label')}
+          validate={null}
+        />
```

Comment:
The same translation-related comments apply here 🙂

Reply:
Fixed in #32